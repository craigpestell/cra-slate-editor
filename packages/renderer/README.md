# @canner/slate-editor-renderer

[![npm version](https://badge.fury.io/js/%40canner%2Fslate-editor-renderer.svg)](https://badge.fury.io/js/%40canner%2Fslate-editor-renderer)

**NOTE: You should directly use it's plugin instead of writing render function.**

```js
import renderer from "@canner/slate-editor-renderer";
```

Renderers that define corresponding nodes and marks while rendering your editor.

methods:

### commonNode(TagName) => React.Element<TagName>

If you want to render general tag names for nodes such as `p`, `blockquote`, `h1`, `h2`, etc...

### commonMark(TagName, styleKey?, styleValue?)

If you want to render general tag names for nodes such as `span`, `strong`, `code`, `i`, etc...

### emojiNode()

rendering emoji

### imageNode()

rendering video

### linkNode()

renderling inline links

### videoNode(source: 'youtube' | 'dailymotion' | 'youku' | 'vimeo')

rendering video

Supported video sources:

* youtube
* dailymotion
* youku
* vimeo

### Complete example:

```js
// all cases
function renderMark(props) {
  // pass this function to your editor's renderMark prop
  switch (props.mark.type) {
    case "bold":
      return commonMark("strong")(props);
    case "code":
      return commonMark("code")(props);
    case "fontBgColor":
      return commonMark("span", "backgroundColor", "color")(props);
    case "fontColor":
      return commonMark("span", "color", "color")(props);
    case "fontSize":
      return commonMark("span", "fontSize")(props);
    case "letterSpacing":
      return commonMark("span", "letterSpacing")(props);
    case "italic":
      return commonMark("i")(props);
    case "strikethrough":
      return commonMark("s")(props);
    case "underline":
      return commonMark("u")(props);
  }
}

function renderNode(props) {
  // pass this function to your editor's renderNode prop
  switch (props.node.type) {
    case "paragraph":
      return commonNode("p")(props);
    case "blockquote":
      return commonNode("blockquote")(props);
    case "emoji":
      return emojiNode()(props);
    case "heading1":
      return commonNode("h1")(props);
    case "heading2":
      return commonNode("h2")(props);
    case "list-ul":
      return commonNode("ul")(props);
    case "list-ol":
      return commonNode("ol")(props);
    case "list-item":
      return commonNode("li")(props);
    case "image":
      return imageNode()(props);
    case "link":
      return linkNode()(props);
    case "youtube":
      return videoNode("youtube")(props);
    case "dailymotion":
      return videoNode("dailymotion")(props);
    case "youku":
      return videoNode("youku")(props);
    case "vimeo":
      return videoNode("vimeo")(props);
  }
}
```

# @canner/slate-editor-html

Serializer and deserializer
