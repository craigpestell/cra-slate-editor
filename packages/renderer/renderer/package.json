{
  "name": "@canner/slate-editor-renderer",
  "version": "1.5.0",
  "description": "Renderers for slate nodes and marks",
  "main": "lib/index.js",
  "author": "",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "peerDependencies": {
    "react": "^0.14.0 || ^15.0.0 || 16.x",
    "react-dom": "^0.14.0 || ^15.0.0 || 16.x",
    "slate": "^0.33.0",
    "styled-components": "^3.2.1"
  },
  "dependencies": {
    "@canner/slate-helper-inline-adddata": "^1.5.0",
    "emoji-mart": "^2.5.0",
    "js-video-url-parser": "^0.2.7",
    "lodash.mapvalues": "^4.6.0",
    "prismjs": "^1.15.0",
    "quill-icons": "^1.5.0",
    "react-icons": "^2.2.7",
    "react-loading-image": "^0.5.0",
    "react-resizable": "^1.7.5",
    "slate-edit-table": "^0.15.0-alpha.4"
  },
  "gitHead": "bb30f8746be7cbe2cda641b6b74c9d976bb56e31"
}
