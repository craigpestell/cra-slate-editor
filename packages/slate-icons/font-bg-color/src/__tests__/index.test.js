import React from "react";
import { Change, Value } from "slate";
import FontBgColor from "../";
import renderer from "react-test-renderer";

test("create a FontBgColor icon", () => {
  const initialValue = Value.fromJSON({
    document: {
      nodes: [
        {
          object: "block",
          type: "paragraph",
          nodes: [
            {
              object: "text",
              leaves: [
                {
                  text: "A line of text in a paragraph."
                }
              ]
            }
          ]
        }
      ]
    }
  });

  const component = renderer.create(
    <FontBgColor change={new Change({ value: initialValue })} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
