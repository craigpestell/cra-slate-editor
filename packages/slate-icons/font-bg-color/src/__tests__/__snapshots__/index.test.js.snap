// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create a FontBgColor icon 1`] = `
<div
  className="rc-color-picker-wrap "
>
  <span
    data-active={false}
    onClick={[Function]}
    onMouseDown={[Function]}
  >
    <svg
      className={undefined}
      height="18px"
      onChange={undefined}
      viewBox="0 0 18 18"
      width="18px"
    >
      <g
        className="ql-fill ql-color-label"
        style={Object {}}
      >
        <polygon
          points="6 6.868 6 6 5 6 5 7 5.942 7 6 6.868"
        />
        <rect
          height="1"
          width="1"
          x="4"
          y="4"
        />
        <polygon
          points="6.817 5 6 5 6 6 6.38 6 6.817 5"
        />
        <rect
          height="1"
          width="1"
          x="2"
          y="6"
        />
        <rect
          height="1"
          width="1"
          x="3"
          y="5"
        />
        <rect
          height="1"
          width="1"
          x="4"
          y="7"
        />
        <polygon
          points="4 11.439 4 11 3 11 3 12 3.755 12 4 11.439"
        />
        <rect
          height="1"
          width="1"
          x="2"
          y="12"
        />
        <rect
          height="1"
          width="1"
          x="2"
          y="9"
        />
        <rect
          height="1"
          width="1"
          x="2"
          y="15"
        />
        <polygon
          points="4.63 10 4 10 4 11 4.192 11 4.63 10"
        />
        <rect
          height="1"
          width="1"
          x="3"
          y="8"
        />
        <path
          d="M10.832,4.2L11,4.582V4H10.708A1.948,1.948,0,0,1,10.832,4.2Z"
        />
        <path
          d="M7,4.582L7.168,4.2A1.929,1.929,0,0,1,7.292,4H7V4.582Z"
        />
        <path
          d="M8,13H7.683l-0.351.8a1.933,1.933,0,0,1-.124.2H8V13Z"
        />
        <rect
          height="1"
          width="1"
          x="12"
          y="2"
        />
        <rect
          height="1"
          width="1"
          x="11"
          y="3"
        />
        <path
          d="M9,3H8V3.282A1.985,1.985,0,0,1,9,3Z"
        />
        <rect
          height="1"
          width="1"
          x="2"
          y="3"
        />
        <rect
          height="1"
          width="1"
          x="6"
          y="2"
        />
        <rect
          height="1"
          width="1"
          x="3"
          y="2"
        />
        <rect
          height="1"
          width="1"
          x="5"
          y="3"
        />
        <rect
          height="1"
          width="1"
          x="9"
          y="2"
        />
        <rect
          height="1"
          width="1"
          x="15"
          y="14"
        />
        <polygon
          points="13.447 10.174 13.469 10.225 13.472 10.232 13.808 11 14 11 14 10 13.37 10 13.447 10.174"
        />
        <rect
          height="1"
          width="1"
          x="13"
          y="7"
        />
        <rect
          height="1"
          width="1"
          x="15"
          y="5"
        />
        <rect
          height="1"
          width="1"
          x="14"
          y="6"
        />
        <rect
          height="1"
          width="1"
          x="15"
          y="8"
        />
        <rect
          height="1"
          width="1"
          x="14"
          y="9"
        />
        <path
          d="M3.775,14H3v1H4V14.314A1.97,1.97,0,0,1,3.775,14Z"
        />
        <rect
          height="1"
          width="1"
          x="14"
          y="3"
        />
        <polygon
          points="12 6.868 12 6 11.62 6 12 6.868"
        />
        <rect
          height="1"
          width="1"
          x="15"
          y="2"
        />
        <rect
          height="1"
          width="1"
          x="12"
          y="5"
        />
        <rect
          height="1"
          width="1"
          x="13"
          y="4"
        />
        <polygon
          points="12.933 9 13 9 13 8 12.495 8 12.933 9"
        />
        <rect
          height="1"
          width="1"
          x="9"
          y="14"
        />
        <rect
          height="1"
          width="1"
          x="8"
          y="15"
        />
        <path
          d="M6,14.926V15H7V14.316A1.993,1.993,0,0,1,6,14.926Z"
        />
        <rect
          height="1"
          width="1"
          x="5"
          y="15"
        />
        <path
          d="M10.668,13.8L10.317,13H10v1h0.792A1.947,1.947,0,0,1,10.668,13.8Z"
        />
        <rect
          height="1"
          width="1"
          x="11"
          y="15"
        />
        <path
          d="M14.332,12.2a1.99,1.99,0,0,1,.166.8H15V12H14.245Z"
        />
        <rect
          height="1"
          width="1"
          x="14"
          y="15"
        />
        <rect
          height="1"
          width="1"
          x="15"
          y="11"
        />
      </g>
      <polyline
        className="ql-stroke"
        points="5.5 13 9 5 12.5 13"
      />
      <line
        className="ql-stroke"
        x1="11.63"
        x2="6.38"
        y1="11"
        y2="11"
      />
    </svg>
  </span>
</div>
`;
